### PR TITLE
Ruby 3 Upgrade 

### PR DESCRIPTION
**Pull request by rawOrlando**
Todos:
- [x] Decide which Ruby version 3
  - 3.2.2?

- [x] List deppendencies that are not compataiable
  Blacklight
  - [x] Rails
     - [ ] Rails app:update
  - [x] Blacklight
  - [x] active-fedora
  has to be 14.x.x to be compatible with ruby 3 and rails 7

- [x] Get Ui looking right again
  - [x] Fix Facets limits
  - [x] Facets use css again
  - [x] Facets Year range broken
  - [x] Help Page Markdown stopped working
  - [x] Padding Issue Search Bar
  - [x] Pagination looks funny now
- [x] Fix tests
    - [x] ci test issues (ERROR: Error CREATEing SolrCore 'hydra-test': Unable to create core [hydra-test] Caused by: solr.IntPointField)
    - [x] docker test issues (Failed to execute solr start: WARNING: Starting Solr as the root user is a security risk and not considered best practice. Exiting.)
    - [x] test failures
        - [x] rspec ./spec/controllers/candidates_controller_spec.rb:5 # CandidatesController should have index
        - [x] catalog controller 404ing
        - [x]  CandidateHelper

- [x] Link from Candidate Helper is the wrong link (grant /track/ at end of link)
- [ ] 

- [x] Different search results?
       Some filter thing broke in (catalog_controller?).
        - [ ] Filters are not getting applied at all 
                - [ ] Filter for displays in not getting applied in index(?)

### CODE REVIEW

File: app/models/solr_document.rb [hunk 1]
```diff
@@ -16,10 +16,4 @@ class SolrDocument
   # and Blacklight::Document::SemanticFields#to_semantic_values
   # Recommendation: Use field names from Dublin Core
   use_extension(Blacklight::Document::DublinCore)
-  field_semantics.merge!(
-                         title: "title_tesi",
-                         author: "author_tesim",
-                         language: "language_ssim",
-                         format: "format_ssim"
-                       )
 end
```
Comment (rawOrlando) [inline app/models/solr_document.rb:19]: I am not sure why I removed this? @mkorcy you helped with a solo issue. was this you?

File: .solr_wrapper.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Place any default configuration for solr_wrapper here
```
Comment (mkorcy) [inline .solr_wrapper.yml:1]: This file now gives the project 2 solr_wrappers i don't know which one is actually getting referenced here, or if either of them get referenced.  I suspect they can both be deleted. 2nd one is in config/ should clean it up for clarity while we're doing this.

**Review by mkorcy — CHANGES_REQUESTED** [review]
I made some other comments most notably about cleaning up the solr_wrapper config but htis mostly looks good to go pending that cleanup

File: app/helpers/candidate_helper.rb [hunk 3]
```diff
@@ -1,20 +1,18 @@
 # frozen_string_literal: true
+
 module CandidateHelper
-  include Blacklight::SearchHelper
+  def related_elections
+    new_search_service = search_service_class.new(config: blacklight_config, user_params: {
```
[the remark this answers is not in the document]
Comment (rawOrlando) [reply app/helpers/candidate_helper.rb]: This was a vestige of me debug the I had a bunch of search services, I changed the name to drop the `new`

File: app/models/solr_document.rb (hunk 1 above)
Comment (rawOrlando) [reply app/models/solr_document.rb]: I think I regenerated this file at some point. Adding this code back

**Comment by rawOrlando** [comment]
How I plan to update ruby version
`
ssh elections-dev-01.lib.tufts.edu -l rorlan02`

# Install ruby 3.2.2
```
sudo /bin/su - hydradm

rvm install 3.2.2

rvm --default use 3.2.2

rvm use 3.2.2
```

# Change configs
cd /etc/httpd/conf.d

```
vim elections_ssl.conf
vim elections.conf
```

And change ruby version

# Change Jenkin files
Like here https://gitlab.it.tufts.edu/tdr/nnv-deployment/-/merge_requests/1/diffs

**Review by mkorcy — APPROVED** [review]
👍

**Comment by rawOrlando** [comment]
Issues experienced during the the upgrade installing Ruby 3 with Rvm

Could not rvm to recognize ruby 3 as existing 

Fix
Get stable version:
`rvm get stable`
Get new keys:
`command curl -sSL https://rvm.io/pkuczynski.asc | gpg --import -`

Ssl error 
reinstall ssl
`rvm pkg install openssl`
Install with specific openssl
`rvm install 3.2.2 -C --with-openssl-dir=/home/hydradm/.rvm/usr`

The `rvm get head` output from will show you where rvm is installed. use that path for about /usr

**Pull request merged**